### PR TITLE
Add proper interpretation of boolean values when strings.

### PR DESCRIPTION
**Pull request by doctoryes**
Added utility function for translating string "boolean" values from the configs to Python booleans.
Changed the existing places where the code interprets those strings.

Companion PR to this revert: https://github.com/edx-ops/gomatic-secure/pull/131

### CODE REVIEW

File: edxpipelines/utils.py [hunk 1]
```diff
@@ -30,6 +30,14 @@ class MergeConflict(Exception):
 EDP.__new__.__defaults__ = (None, None)
 
 
+def as_bool(value):
+    """
+    Return the bool value for 'value'.
+    Translates a string value to a Python boolean. Allows all valid YAML boolean values.
+    """
+    return str(value).lower() in ('true', 'y', 'yes', 'on')
```
Comment (jibsheet) [inline edxpipelines/utils.py]: what, no "Affirmative" ?

Comment (doctoryes) [inline edxpipelines/utils.py]: 😸  - I used this official definition.

Comment (doctoryes) [inline edxpipelines/utils.py]: http://yaml.org/type/bool.html

Comment (pwnage101) [inline edxpipelines/utils.py]: a-o-k
right-o
yep
uh huh

Comment (jibsheet) [inline edxpipelines/utils.py]: I'm honestly surprised Ingy didn't try to shove more nonsense into the spec, seems his style

**Pull request merged**